src/VideoNativeComponent.ts

```diff
@@ -0,0 +1,323 @@
+import type { HostComponent, NativeSyntheticEvent, ViewProps } from 'react-native';
+import { NativeModules, requireNativeComponent } from 'react-native';
+import { getViewManagerConfig } from './utils';
+
+// -------- There are types for native component (future codegen) --------
+// if you are looking for types for react component, see src/types/video.ts
+
+type Headers = Record<string, any>
+
+type VideoSrc = Readonly<{
+  uri?: string;
+  isNetwork?: boolean;
+  isAsset?: boolean;
+  shouldCache?: boolean;
+  type?: string;
+  mainVer?: number;
+  patchVer?: number;
+  requestHeaders?: Headers;
+  startTime?: number;
+  endTime?: number;
+}>
+
+
+export type Filter = 'None' |
+'CIColorInvert' |
+'CIColorMonochrome' |
+'CIColorPosterize' |
+'CIFalseColor' |
+'CIMaximumComponent' |
+'CIMinimumComponent' |
+'CIPhotoEffectChrome' |
+'CIPhotoEffectFade' |
+'CIPhotoEffectInstant' |
+'CIPhotoEffectMono' |
+'CIPhotoEffectNoir' |
+'CIPhotoEffectProcess' |
+'CIPhotoEffectTonal' |
+'CIPhotoEffectTransfer' |
+'CISepiaTone';
+
+export type DrmType = 'widevine' | 'playready' | 'clearkey' | 'fairplay';
+
+type Drm = Readonly<{
+  drmType?: DrmType;
+  licenseServer?: string;
+  headers?: Headers;
+  contentId?: string; // ios
+  certificateUrl?: string; // ios
+  base64Certificate?: boolean; // ios default: false
+  useExternalGetLicense?: boolean; // ios
+}>
+
+type TextTracks = ReadonlyArray<Readonly<{
+  title: string;
+  language: string;
+  type: string;
+  uri: string;
+}>>
+
+type TextTrackType = 'system' | 'disabled' | 'title' | 'language' | 'index';
+
+type SelectedTextTrack = Readonly<{
+  selectedTextType?: TextTrackType;
+  value?: string;
+  index?: number;
+}>
+
+type AudioTrackType = 'system' | 'disabled' | 'title' | 'language' | 'index';
+
+type SelectedAudioTrack = Readonly<{
+  selectedAudioType?: AudioTrackType;
+  value?: string;
+  index?: number;
+}>
+
+export type Seek = Readonly<{
+  time: number;
+  tolerance?: number;
+}>
+
+type BufferConfig = Readonly<{
+  minBufferMs?: number;
+  maxBufferMs?: number;
+  bufferForPlaybackMs?: number;
+  bufferForPlaybackAfterRebufferMs?: number;
+  maxHeapAllocationPercent?: number;
+  minBackBufferMemoryReservePercent?: number;
+  minBufferMemoryReservePercent?: number;
+}>
+
+type SelectedVideoTrack = Readonly<{
+  type?: 'auto' | 'disabled' | 'resolution' | 'index';
+  value?: number;
+}>
+
+type SubtitleStyle = Readonly<{
+  fontSize?: number;
+  paddingTop?: number;
+  paddingBottom?: number;
+  paddingLeft?: number;
+  paddingRight?: number;
+}>
+
+export type OnLoadData = Readonly<{
+  currentTime: number;
+  duration: number;
+  naturalSize: Readonly<{
+    width: number;
+    height: number;
+    orientation: string;
```

Comment:
There is no enum for orientation? That is sad 😢

Reply:
Added `'portrait' | 'landscape'` for now